core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,209 @@
+            setAclsRecursively(zkUtils, path)
+          case Code.NONODE =>
+            warn("Node is gone, it could be have been legitimately deleted: %s".format(path))
+          case Code.SESSIONEXPIRED =>
+            // Starting a new session isn't really a problem, but it'd complicate
+            // the logic of the tool, so we quit and let the user re-run it.
+            error("ZooKeeper session expired while changing ACLs")
+            exception = throw ZkException.create(KeeperException.create(rc))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            exception = throw ZkException.create(KeeperException.create(rc))
+        }
+      }
+  }
+  
+  object SetACLCallback extends StatCallback {
+    def processResult(rc: Int,
+                      path: String,
+                      ctx: Object,
+                      stat: Stat) {
+      val zkUtils: ZkUtils = ctx.asInstanceOf[ZkUtils]
+      val zkHandle = zkUtils.zkConnection.getZookeeper
+        
+      Code.get(rc) match {
+          case Code.OK =>
+            info("Successfully set ACLs for %s".format(path))
+            counter.decrementAndGet()
+          case Code.CONNECTIONLOSS =>
+            Future {
+              zkHandle.setACL(path, ZkUtils.DefaultAcls(true), -1, SetACLCallback, ctx)
+            }(threadExecutionContext)
+          case Code.NONODE =>
+            warn("Node is gone, it could be have been legitimately deleted: %s".format(path))
+          case Code.SESSIONEXPIRED =>
+            // Starting a new session isn't really a problem, but it'd complicate
+            // the logic of the tool, so we quit and let the user re-run it.
+            error("ZooKeeper session expired while changing ACLs")
+            exception = throw ZkException.create(KeeperException.create(rc))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            exception = throw ZkException.create(KeeperException.create(rc))   
+      }
+    }
+  }
+  
+  def run(args: Array[String]) {
+    var jaasFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
+    val parser = new OptionParser()
+
+    val jaasFileOpt = parser.accepts("jaas.file", "JAAS Config file.").withOptionalArg().ofType(classOf[String])
+    val zkUrlOpt = parser.accepts("connect", "Sets the ZooKeeper connect string (ensemble). This parameter " +
+                                  "takes a comma-separated list of host:port pairs.").
+      withRequiredArg().defaultsTo("localhost:2181").ofType(classOf[String])
+    val zkSessionTimeoutOpt = parser.accepts("session.timeout", "Sets the ZooKeeper session timeout.").
+      withRequiredArg().ofType(classOf[java.lang.Integer])
+    val zkConnectionTimeoutOpt = parser.accepts("connection.timeout", "Sets the ZooKeeper connection timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val helpOpt = parser.accepts("help", "Print usage information.")
```

Comment:
To support going from secured ZK to unsecured ZK, we will need an option to enable/disable the CREATOR_ALL acl.